src/audio/mux/mux.c

```diff
@@ -200,7 +204,195 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
 	dev->state = COMP_STATE_READY;
 	return dev;
 }
+#else
+static int build_config(struct comp_data *cd, struct comp_dev *dev)
+{
+	dev->ipc_config.type = SOF_COMP_MUX;
+	cd->config.frame_format_deprecated = cd->md.base_cfg.audio_fmt.depth;
+	cd->config.num_channels_deprecated = cd->md.base_cfg.audio_fmt.channels_count;
+	cd->config.num_streams = 2;
+	/* clear masks */
+	for (uint16_t i = 0; i < cd->config.num_streams; i++)
+		for (uint16_t j = 0 ; j < PLATFORM_MAX_CHANNELS; j++)
+			cd->config.streams[i].mask[j] = 0;
+	int mask = 1;
```

Comment:
Declare variable at start of scope.

Reply:
This is not required since C99 (or any other modern programming language) and it restricts the use of `const`.

We had places where declarations are not at the top of the block (#3995) and they never caused any issue with any compiler.